### ISSUE TITLE
Training the fifo model

### ISSUE THREAD

**Issue by yimuwangcs** [sohyun-l/fifo#3]
Hi, 

Thanks for sharing the code of this valuable work.

I was using the following command to train the FIFO model.
`
python3 main.py --file-name 'FIFO_model'   --restore-from './Cityscapes_pretrained_model.pth' --restore-from-fogpass './FogPassFilter_pretrained.pth' --modeltrain 'train' 
`

But I found the performance is not that good.  I only get the following results.

Evaluation on Foggy Zurich
===> mIoU: 40.99
Evaluation on Foggy Driving Dense
===> mIoU: 37.74
Evaluation on Foggy Driving
===> mIoU: 47.55
Evaluation on Cityscapes lindau 40
===> mIoU: 64.85

Are there any other tricks for training the model or selecting the model?

Best

**Comment by DhruvaRajwade** [comment]
Hi, I was also trying to run on a 16 gb GPU, but was facing OOM issues. What specs were you able to run the code on? Just needed a reference config

**Comment by sohyun-l** [comment]
Hi, @yimuwangcs 
Thanks for your interest in our work!
We also found that if the evaluation is performed on datasets that were not used during training, there may be a variance in performance, similar to existing domain generalization models. We minimize the variance by fixing the initial random seed, hardware, and environment settings.
If you want to check the result of our final model, we recommend you utilize our pre-trained FIFO model weights.

**Comment by sohyun-l** [comment]
Hi, @DhruvaRajwade 
Thanks for your interest in our paper, FIFO!
We train the FIFO model with a single RTX 3090 GPU (24GB).

**Comment by yimuwangcs** [comment]
> Hi, @yimuwangcs
> Thanks for your interest in our work!
> We also found that if the evaluation is performed on datasets that were not used during training, there may be a variance in performance, similar to existing domain generalization models. We minimize the variance by fixing the initial random seed, hardware, and environment settings.
> If you want to check the result of our final model, we recommend you utilize our pre-trained FIFO model weights.

Hi, thanks for your reply!

I have tried this. But the final results are not good as in the paper. So I was wondering if you have any specific training tricks? such as early stopping or selection methods.

Best regards

**Comment by sohyun-l** [comment]
Hi, we do not have any additional training techniques to offer besides the code that has been published.  
We recommend utilizing the pre-trained weights of the fog-pass filtering module that we have trained.
